refactor: Cleanup services for deep linking

Zeyber opened the pull request.

scarai-sap committed:
CXSPA-3113 As an assisted service agent, I want to open deep link for Spartacus containing customerID only so that the emulation session can be started automatically for the customer (#17345)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Hak Woo Kim <kimhakwoo@hotmail.com>
Co-authored-by: Caine Rotherham <rotherham.c@gmail.com>

scarai-sap committed:
CXSPA-3113: Fix e2e test failure (#17414)

Zeyber committed:
feat: ASM Deep Links for ticketId, orderId and savedCartId (#17440)

Closes: https://jira.tools.sap/browse/CXSPA-3263

i333055 committed:
feat: CXSPA-3278 As an assisted service agent, I want to open deep link for Spartacus containing inactiveCartId  (#17380)

Co-authored-by: Scar Ai <scar.ai@sap.com>
Co-authored-by: Scar Ai <38693320+scarai-sap@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Hak Woo Kim <kimhakwoo@hotmail.com>
Co-authored-by: Caine Rotherham <rotherham.c@gmail.com>

niehuayang committed:
fix: handling the backend error of Creating New Customer  and Improving UI (#17428)

Zeyber committed:
Merge branch 'develop-6.1.x' into epic/CXSPA-158

Zeyber committed:
fix build libs failure by exporting save cart dialog

Zeyber committed:
fix sonar error about too many providers in constructor

i333055 committed:
fix: bugfix for inactive cart deeplink (#17457)

Zeyber committed:
correct deprecation constructor on main component

Zeyber committed:
Merge branch 'epic/CXSPA-158' of https://github.com/SAP/spartacus into epic/CXSPA-158

Zeyber committed:
comment out deprecation to see if sonar passes

Zeyber committed:
fix typescript build error

Zeyber committed:
remove deprecation note on isEmulateInURL

Zeyber committed:
fix es-lint error on constructor overload

Zeyber committed:
fix prettier error and asm component service unit tests

i333055 committed:
feat: CXSPA-3507 code for deeplink active cart (#17467)

i333055 committed:
fix: bug fix for deeplink  (#17474)

i333055 committed:
Merge branch 'develop-6.2.x' into epic/CXSPA-158

i333055 committed:
Merge branch 'develop-6.2.x' into epic/CXSPA-158

Zeyber committed:
Merge branch 'develop-6.2.x' into epic/CXSPA-158

Zeyber committed:
refactor: Cleanup services for deep linking

github-actions[bot] committed:
Add license header

Zeyber committed:
remove tests

Zeyber committed:
Merge branch 'feature/CXSPA-158-cleanup' of https://github.com/SAP/spartacus into feature/CXSPA-158-cleanup

Zeyber committed:
move back some sub logic

github-actions[bot] committed:
Add license header

Zeyber committed:
more clean up

Zeyber committed:
fix merge

github-actions[bot] committed:
Add license header

i333055 commented on feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts:
```diff
@@ -313,39 +314,24 @@ export class AsmBindCartComponent implements OnInit, OnDestroy {
+            const cartType =
+              this.asmComponentService?.getSearchParameter('cartType');
+            if (cartType === 'inactive' || cartType === 'active') {
+              this.displayBindCartBtn$.next(cartType === 'inactive');
```
```suggestion
              this.displayBindCartBtn$.next(false);
```

i333055 commented on feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts:
```diff
@@ -313,39 +314,24 @@ export class AsmBindCartComponent implements OnInit, OnDestroy {
+              this.asmComponentService?.getSearchParameter('cartType');
+            if (cartType === 'inactive' || cartType === 'active') {
+              this.displayBindCartBtn$.next(cartType === 'inactive');
               this.displaySaveCartBtn$.next(true);
```
```suggestion
              this.displaySaveCartBtn$.next(cartType === 'inactive');
```

Zeyber committed:
fix unit tests

Zeyber committed:
fix sonar issues

scarai-sap commented on feature-libs/asm/root/services/asm-deep-link.service.ts:
```diff
@@ -0,0 +1,108 @@
+
+    const ALLOWED_PARAMS = [
+      ['customerId'],
+      ['customerId', 'cartType'],
```
we don't support cartType without cartId,

We need support ticketid + orderId/ ticketId + cartId + cartType

Zeyber replied:
What is the expected behaviour happen when `ticketid + orderId` or `ticketId + cartId + cartType ` are in the url? Isn't having both invalid?

scarai-sap commented on feature-libs/asm/root/services/asm-deep-link.service.ts:
```diff
@@ -0,0 +1,108 @@
+   * Returns true if pairs of valid navigation params are in the url only.
+   * eg. customerId, customerId + ticketId return true.
+   */
+  validParameters(parameters: AsmDeepLinkParameters): boolean {
```
why we need here, as in caller "handleNavigation" has enough check as if.. elif.. elif

It will block if other apps, consider price domain want to show mass prices with property with a flag "displayMassPrices", a url like below maybe composed, we should support navigate

cartId=123&displayMassPrices=true

Zeyber replied:
Good point. We shouldn't have additional parameters preventing a navigation. However, the `getParamsInUrl()` method only checks for `AsmDeepLinkParameters` deep link params specifically and additional parameters will not be compared when using this method (ie. has no effect on navigation).

Zeyber committed:
add units

Zeyber committed:
add todo for units

RadhepS committed:
Release 6.2.0-1

sisodiyam committed:
CXINT-1833: added the cypress test for the register organization (#17505)

Co-authored-by: anjana-bl <anjana.b.l@sap.com>

niehuayang committed:
refine busy icon diaplay for ASM crerate new customer (#17533)

steinsebastian committed:
fix: accessibilty for read-only attributes (#17537)

i333055 committed:
fix: display confirm switch customer dialog when new customerId in the deeplink (#17530)

Co-authored-by: Scar Ai <38693320+scarai-sap@users.noreply.github.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Hak Woo Kim <kimhakwoo@hotmail.com>
Co-authored-by: Caine Rotherham <rotherham.c@gmail.com>
Co-authored-by: Scar Ai <scar.ai@sap.com>
Co-authored-by: niehuayang <126219214+niehuayang@users.noreply.github.com>
Co-authored-by: Giancarlo Cordero Ortiz <46171897+giancorderoortiz@users.noreply.github.com>

anjana-bl committed:
cdc consent management - review changes (#17546)

i333055 committed:
fix: add jira num for e2e tests (#17552)

luoyifeng11 committed:
test: Fix the instability issue in ASM E2E tests, as well as the failure issue in asm-cost-center.e2e (#17487)

pawelfras committed:
feat: CXSPA-3742 - SSR logging - addresing minor comments (#17538)

scarai-sap committed:
feat: update text as UX suggested #17522 for 6.2 (#17565)

luoyifeng11 committed:
test: fix ASM E2E test instability for 6.2 (#17569)

i53577 committed:
fix: Need to adapt to the width of mobile webpage on CustomerList (#17576)

niehuayang committed:
fix/CXSPA-3848: cherry-pick buf fixes from branch 6.1.x to branch 6.2.x (#17579)

ChristophHi committed:
fix: configurator restart must remove UI state as well (#17581)

See https://jira.tools.sap/browse/CXSPA-3839

pawelfras committed:
chore: CXSPA-3794 - SSR contextual logging extensibility (#17578)

Platonn committed:
fix: 'this' is undefined in ProductEffects.productLoadEffect (#17594)

Previous behavior: When `/products` endpoint returned a http error, the code broke in [this line](https://github.com/SAP/spartacus/blob/ed1e1a78c488b1e1214491ffa736612287f8cf70/projects/core/src/product/store/effects/product.effect.ts#L77), complaining that `this` is undefined.

Fix: Preserve the context of `this` which was lost in [this line](https://github.com/SAP/spartacus/blob/ed1e1a78c488b1e1214491ffa736612287f8cf70/projects/core/src/product/store/effects/product.effect.ts#L52)

The problem was revealed only after we implemented [CXSPA-2251](https://jira.tools.sap/browse/CXSPA-2251) where we referenced `this` by adding `this.logger` to the method `ProductEffects.productLoadEffect`

fixes https://jira.tools.sap/browse/CXSPA-3902

Zeyber committed:
merge

Zeyber committed:
Merge branch 'develop-6.3.x' into feature/CXSPA-158-cleanup

Zeyber committed:
cleanup merge errors

Zeyber committed:
fix sonar

Zeyber committed:
fix navigation issues

Zeyber committed:
cleanup main ui

Zeyber committed:
cleanup main ui

Zeyber committed:
fix unit tests

Zeyber committed:
fix switch customer unit test

Zeyber committed:
fix asm component service

Zeyber committed:
fix circular dep

Zeyber committed:
add dep ticket

Zeyber committed:
add unit tests

Zeyber committed:
Merge branch 'develop-6.3.x' into feature/CXSPA-158-cleanup

i333055 committed:
Merge branch 'develop-6.3.x' into feature/CXSPA-158-cleanup

Zeyber committed:
Merge branch 'develop-6.3.x' into feature/CXSPA-158-cleanup
